woocommerce/woocommerce-ios
# Support form: Handle creating identity without UIKit

Closes: #11758 


## Description

Previously, we required using `SupportFormHostingController` to handle creating an identity for the support form. `HostedSupportForm` was created after that to add a SwiftUI wrapper around `SupportFormHostingController,` but this is not a nice solution and has some navigation glitches around the identity input alert.

This PR updates the support form and `ZendeskManager` to get contact details without the requirement for UIKit and marks `HostedSupportForm` as deprecated.

Changes:
- Updated `ZendeskManager`:
  - Added a new async method `createIdentity`.
  - Made `haveUserIdentity` accessible.
  - Replaced `userSupportEmail` with `userSupportIdentity`.
  - Added new error type `ZendeskError`.
- SupportForm:
  - Added an option closure `onDismiss` and update dismiss handling for `SupportFormHostingController`.
  - Added alerts for success, failure, and identity input.
- SupportFormViewModel:
  -  Removed `onCompletion`.
  - Added `shouldHandleIdentity` to handle identity input.
- BlazeCampaignCreationErrorView: Update the latest integration for `SupportForm`.

## Testing instructions

Testing from a SwiftUI view:
- Update the method submitCampaign in BlazeConfirmPaymentViewModel to force shouldDisplayCampaignCreationError to be true and build the app.
- Log out of the store if you already entered an email address for the support form before.
- Log in to a store that's eligible for Blaze.
- Select Promote in the Blaze section on the dashboard screen to start campaign creation.
- Select Confirm Details > Submit Campaign.
- When the error screen is displayed, tap Get Support.
- Notice an alert is displayed for entering name and email address. Tap Cancel, the support form should be dismissed.
- Tap Get Support again. Enter a non-A8C email address for this alert. Notice that the OK button is only enabled when a correct form of the address is entered.
- Fill the support form. Please note in the request that this is a test ticket and it should be closed.
- Disable the Internet connection on your machine and send the request. Notice an error alert after the request fails.
- Enable the connection and try again. After the request succeeds, notice an alert for the success. Tapping Got It should dismiss the support form.

Testing from a UIKit view:
- Navigate to the Menu tab.
- Select Subscriptions > Report subscription issue.
- Repeat the testing steps for the support form to confirm that everything works exactly the same.

## Screenshots

 

Identity Input | Success | Failure
--- | --- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f8617115-af34-4025-865b-fb5904920403" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/ea07923c-5824-42f0-8c30-583d2cb57be9" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/c9b790d6-6e24-4d2e-84e1-04d4b42835dc" width=320 />

---
